### ISSUE TITLE
AddJsonOptions does not seem to be updating the output format

### ISSUE THREAD

**Issue by preetriti1** [aspnet/AspNetCore#15093]
### Describe the bug
Unable to add custom json serialization to the output of web app controllers

### To Reproduce
1. Using this version of ASP.NET Core '2.2'
2. Run this code 
 I have added the below in my ConfigureServices in Startup.cs
   `
 services
                .AddMvc(options =>
                {
                    options.Filters.Add(typeof(ErrorResponseFilter));
                    options.EnableEndpointRouting = true;
                })
                .AddJsonOptions(options =>
                {
                    options.SerializerSettings.ContractResolver = new CamelCasePropertyNamesContractResolver();
                })
                .SetCompatibilityVersion(CompatibilityVersion.Version_2_2)
                .AddWebApiConventions();
`

and then also have this in my Startup.cs

`
var formatters = new MediaTypeFormatterCollection();
            formatters.Add(FlowJsonExtensions.JsonMediaTypeFormatter);
            services.AddSingleton<MediaTypeFormatterCollection>(formatters);
`

3. With these arguments '....'
4. See error - There is no error

### Expected behavior
The return response needs to follow the json serialization i have added, but all responses are in the usual format as -
{ Id: "abc", Value: Null }
expected is { id: "abc" }

### Additional context
Include the output of `dotnet --info`

**Comment by pranavkm** [comment]
@preetriti1 what's `FlowJsonExtensions.JsonMediaTypeFormatter`? 

`AddJsonOptions` specifically configures MVC"s built-in `JsonInputFormatter` \ `JsonOutputFormatter`. If the instance you're adding isn't the same instance but gets uses by the action, it would explain why things are not working. If you think this is an actual issue with MVC, please include a minimal app that reproduces the problem and we can investigate.

**Comment by preetriti1** [comment]
FlowJsonExtensions.JsonMediaTypeFormatter is our own formatter with specific json settings.
We thought adding this in the mediatypeformatter will convert all httpresponsemessages in the specific format, and addJsonOptions was not working when we added the shim conversion.

We fixed the above issue by initializing formatters in the constructor of MediaTypeFormatterCollection and removed AddJsonOptions.
